In [1]:
#Importando as Bibliotecas Necessárias para poder fazer o tratamento e a análise exploratória do dados
import pandas as pd
import numpy as np

In [7]:
#Importando a base dedos e colocando em uma variável
arq = pd.read_csv("HFCRD.csv")

In [8]:
#Visualizando os dados
arq.head()

age  anaemia  creatinine_phosphokinase  diabetes  ejection_fraction  \
0  75.0        0                       582         0                 20   
1  55.0        0                      7861         0                 38   
2  65.0        0                       146         0                 20   
3  50.0        1                       111         0                 20   
4  65.0        1                       160         1                 20   

   high_blood_pressure  platelets  serum_creatinine  serum_sodium  sex  \
0                    1  265000.00               1.9           130    1   
1                    0  263358.03               1.1           136    1   
2                    0  162000.00               1.3           129    1   
3                    0  210000.00               1.9           137    1   
4                    0  327000.00               2.7           116    0   

   smoking  time  DEATH_EVENT  
0        0     4            1  
1        0     6            1  
2        1     7            1  
3        0     7            1  
4        0     8            1

In [9]:
#Verificando a dimensão dos dados
arq.shape

(299, 13)

In [10]:
#Verificando as Variáveis dos dados
arq.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       299 non-null    float64
 1   anaemia                   299 non-null    int64  
 2   creatinine_phosphokinase  299 non-null    int64  
 3   diabetes                  299 non-null    int64  
 4   ejection_fraction         299 non-null    int64  
 5   high_blood_pressure       299 non-null    int64  
 6   platelets                 299 non-null    float64
 7   serum_creatinine          299 non-null    float64
 8   serum_sodium              299 non-null    int64  
 9   sex                       299 non-null    int64  
 10  smoking                   299 non-null    int64  
 11  time                      299 non-null    int64  
 12  DEATH_EVENT               299 non-null    int64  
dtypes: float64(3), int64(10)
memory usage: 30.5 KB


É visto pela função info que o banco de dados não possui informação nula, porém para conformar usarei a função "pd.isna()"

In [12]:
pd.isna(arq).sum()

age                         0
anaemia                     0
creatinine_phosphokinase    0
diabetes                    0
ejection_fraction           0
high_blood_pressure         0
platelets                   0
serum_creatinine            0
serum_sodium                0
sex                         0
smoking                     0
time                        0
DEATH_EVENT                 0
dtype: int64

In [54]:
arq.describe()

age     anaemia  creatinine_phosphokinase    diabetes  \
count  299.000000  299.000000                299.000000  299.000000   
mean    60.833893    0.431438                581.839465    0.418060   
std     11.894809    0.496107                970.287881    0.494067   
min     40.000000    0.000000                 23.000000    0.000000   
25%     51.000000    0.000000                116.500000    0.000000   
50%     60.000000    0.000000                250.000000    0.000000   
75%     70.000000    1.000000                582.000000    1.000000   
max     95.000000    1.000000               7861.000000    1.000000   

       ejection_fraction  high_blood_pressure      platelets  \
count         299.000000           299.000000     299.000000   
mean           38.083612             0.351171  263358.029264   
std            11.834841             0.478136   97804.236869   
min            14.000000             0.000000   25100.000000   
25%            30.000000             0.000000  212500.000000   
50%            38.000000             0.000000  262000.000000   
75%            45.000000             1.000000  303500.000000   
max            80.000000             1.000000  850000.000000   

       serum_creatinine  serum_sodium         sex    smoking        time  \
count         299.00000    299.000000  299.000000  299.00000  299.000000   
mean            1.39388    136.625418    0.648829    0.32107  130.260870   
std             1.03451      4.412477    0.478136    0.46767   77.614208   
min             0.50000    113.000000    0.000000    0.00000    4.000000   
25%             0.90000    134.000000    0.000000    0.00000   73.000000   
50%             1.10000    137.000000    1.000000    0.00000  115.000000   
75%             1.40000    140.000000    1.000000    1.00000  203.000000   
max             9.40000    148.000000    1.000000    1.00000  285.000000   

       DEATH_EVENT  
count    299.00000  
mean       0.32107  
std        0.46767  
min        0.00000  
25%        0.00000  
50%        0.00000  
75%        1.00000  
max        1.00000

In [18]:
arq['DEATH_EVENT'].value_counts()

DEATH_EVENT
0    203
1     96
Name: count, dtype: int64

In [15]:
mortos = arq[arq['DEATH_EVENT'] == 1]
vivos = arq[arq['DEATH_EVENT'] == 0]

In [16]:
mortos

age  anaemia  creatinine_phosphokinase  diabetes  ejection_fraction  \
0    75.0        0                       582         0                 20   
1    55.0        0                      7861         0                 38   
2    65.0        0                       146         0                 20   
3    50.0        1                       111         0                 20   
4    65.0        1                       160         1                 20   
..    ...      ...                       ...       ...                ...   
220  73.0        0                       582         0                 20   
230  60.0        0                       166         0                 30   
246  55.0        0                      2017         0                 25   
262  65.0        1                       258         1                 25   
266  55.0        0                      1199         0                 20   

     high_blood_pressure  platelets  serum_creatinine  serum_sodium  sex  \
0                      1  265000.00              1.90           130    1   
1                      0  263358.03              1.10           136    1   
2                      0  162000.00              1.30           129    1   
3                      0  210000.00              1.90           137    1   
4                      0  327000.00              2.70           116    0   
..                   ...        ...               ...           ...  ...   
220                    0  263358.03              1.83           134    1   
230                    0   62000.00              1.70           127    0   
246                    0  314000.00              1.10           138    1   
262                    0  198000.00              1.40           129    1   
266                    0  263358.03              1.83           134    1   

     smoking  time  DEATH_EVENT  
0          0     4            1  
1          0     6            1  
2          1     7            1  
3          0     7            1  
4          0     8            1  
..       ...   ...          ...  
220        0   198            1  
230        0   207            1  
246        0   214            1  
262        0   235            1  
266        1   241            1  

[96 rows x 13 columns]

In [17]:
vivos

age  anaemia  creatinine_phosphokinase  diabetes  ejection_fraction  \
14   49.0        1                        80         0                 30   
20   65.0        1                        52         0                 25   
23   53.0        0                        63         1                 60   
33   50.0        1                       159         1                 30   
38   60.0        0                      2656         1                 30   
..    ...      ...                       ...       ...                ...   
294  62.0        0                        61         1                 38   
295  55.0        0                      1820         0                 38   
296  45.0        0                      2060         1                 60   
297  45.0        0                      2413         0                 38   
298  50.0        0                       196         0                 45   

     high_blood_pressure  platelets  serum_creatinine  serum_sodium  sex  \
14                     1   427000.0               1.0           138    0   
20                     1   276000.0               1.3           137    0   
23                     0   368000.0               0.8           135    1   
33                     0   302000.0               1.2           138    0   
38                     0   305000.0               2.3           137    1   
..                   ...        ...               ...           ...  ...   
294                    1   155000.0               1.1           143    1   
295                    0   270000.0               1.2           139    0   
296                    0   742000.0               0.8           138    0   
297                    0   140000.0               1.4           140    1   
298                    0   395000.0               1.6           136    1   

     smoking  time  DEATH_EVENT  
14         0    12            0  
20         0    16            0  
23         0    22            0  
33         0    29            0  
38         0    30            0  
..       ...   ...          ...  
294        1   270            0  
295        0   271            0  
296        0   278            0  
297        1   280            0  
298        1   285            0  

[203 rows x 13 columns]

Perguntas a serem respondidas:

1° Qual percentual de pessoas mortas ?

2° Qual a média de idade dos Mortos ?

3° Qual é o percentual de homens e mulheres mortos ? além disso, qual a média de idade de mortes para homens e mulheres?

4° Qual percentual dos mortos que também possuia diabetes ?

5° Qual foi a idade que mais obteve mortos ?

## R - 1° Pergunta: Qual percentual de pessoas mortas ?

In [19]:
#Calculando o percentual de paciêntes mortos
arq.DEATH_EVENT.value_counts(normalize = True)

DEATH_EVENT
0    0.67893
1    0.32107
Name: proportion, dtype: float64

Com isso, segundo os dados coletados, temos que cerca 32% das pessoas morrem por doenças cardiovasculares.

## R - 2° Pergunta: Qual a média de idade dos Mortos ?

In [23]:
#Calculando a idade média dos mortos
mortos['age'].mean()

65.21528124999999

In [24]:
#Calculando a mediana dos dados
mortos['age'].median()

65.0

Sendo assim, temos, que a idade média de morte para pessoas com doenças cardiovasculares, é de aproximadamente 65 anos, além disso podemos dizer 50% das pessoas mortas que participaram da pesquisa possui no mínimo 65 anos de idade.

## R -  3° Pergunta: Qual é o percentual de homens e mulheres mortos ? além disso, qual a média de idade de mortes para homens e mulheres?

In [39]:
#Colocando a quantidade de mortes de cada sexo em uma variável
mm = mortos['sex'].value_counts()[0]
hh = mortos['sex'].value_counts()[1]

In [40]:
mortos['sex'].value_counts(normalize = True)

sex
1    0.645833
0    0.354167
Name: proportion, dtype: float64

Com isso, Chegamos que o número de homens mortos é quase o dobro do número de mulheres, porém essa estatística leva a uma erro na análise
por isso iremos pegar o número total mortes masculinas e dividir pelo número total de homens, farei a mesma coisa para as mortes femininas 

In [41]:
H = arq['sex'].value_counts()[1]
M = arq['sex'].value_counts()[0]

In [42]:
hh/H

0.31958762886597936

In [43]:
mm/M

0.3238095238095238

Assim, avaliando os percentuais de mortos por sexo, veemos que na verdade a porcentagem de mulheres mortas por doenças cardiovasculares é um pouco maior que o percentual de homens, com isso foi possível ter uma análise mais precisa.

In [46]:
hh/arq.shape[0]

0.20735785953177258

In [47]:
mm/arq.shape[0]

0.11371237458193979

Avaliando o número de mortes por sexo pelo número total de individuos na pesquisa, podemos ver uma disparedade, porém, ela é explicada 
pela diferença de individuos do sexo Mas e Fem.

Conclusão: sendo assim, temos que aproximadamente 32.4% das mulheres morrem por doenças cardiovasculares, enquanto para homens é aproximadamente 32%,
logo, dependendo do nivel de confiança que escolhessemos, poderiamos alegar que o número de mortes de homens e mulheres são iguais.    

## R - 4° Pergunta: Qual percentual dos mortos que também possuia diabetes ?

In [59]:
mortos['diabetes'].value_counts(normalize = True)

diabetes
0    0.583333
1    0.416667
Name: proportion, dtype: float64

In [60]:
mortos['high_blood_pressure'].value_counts(normalize = True)

high_blood_pressure
0    0.59375
1    0.40625
Name: proportion, dtype: float64

In [58]:
arq2 = arq.query(" diabetes == 1 & high_blood_pressure == 1 ")
arq2['DEATH_EVENT'].value_counts(normalize = True)


DEATH_EVENT
0    0.604651
1    0.395349
Name: proportion, dtype: float64

Analisando os dados das pessoas que tinham tanto diabetes como hipertensão, chegamos a uma conclusão de que as pessoas que possuem essas duas doenças tem 40% de chance por morrer de doenças cardiovasculares, segundo os dados da tabela.

e analisando os pacientes que morreram, cerca de 40% deles tinham diabetes.

## R - 5° Pergunta: Qual foi a idade que mais obteve mortos ?

In [89]:
arq.groupby('age')['DEATH_EVENT'].count().sort_values().index[-1]

60.0